src/components/Checkbox/CheckboxField.tsx

```
@@ -56,6 +58,7 @@ export const CheckboxField: FC<CheckboxProps> = React.memo(
             tooltipIcon={tooltipIcon}
             touched={meta.touched}
             error={meta.error}
+            noError={noError}
```

How is `noError` different from `error`? Can't this be inferred?

---

@fabio-silva this is prop for ability to disable the display of an empty block with an error, by default we always indent at the bottom, so that when an error occurs, the rest of the content does not shift, but there are cases when validation is not required and we would like to reduce this indentation

---

I see. But isn't `noError` the same as `!meta.error`?

---

I don't think so, meta.error - tells us whether there is an error or not, if we rewrite this piece of code so that it is meta.error instead of new prop, then we will get a situation where if there is no error - the field has a small indent - when the error appears - all the fields that are below twill shift? because we need a place to draw the error, i.e. yes, in theory we could set meta for the field from the outside, and we can use meta.error to show/hide error block, but in practice if we don't want to get the behavior described above everywhere, we can't use meta.error.I hope that I was able to convey this idea

---

https://web.dev/cls/

---

Ok, I think I get it now, thanks :)

---

in other words,` "noError"` says "I want an empty field for a possible error never to be displayed"
when `!meta.error` in the same usage says "I want an empty field for a possible error not to be displayed when there is no error"